src/libs/react-utils.ts

```diff
@@ -95,7 +97,8 @@ export const combineRefs = refs => {
   }
 }
 
-export const forwardRefWithName = _.curry((name, WrappedComponent) => {
+// TODO: improve these types further
+export const forwardRefWithName = safeCurry((name: string, WrappedComponent) => {
```

Comment:
I would have liked to type `WrappedComponent` here, but unfortunately its parameterized, which causes backwards-incompatible changes for consumers of `fowardRefWithName` that are already in typescript

Reply:
hmmm.... would this work:

```
const forwardRefWithNameFn = <T>(name: string, WrappedComponent: T) => T {
  return withDisplayName(name, forwardRef(WrappedComponent)) as T
}
export const forwardRefWithName = safeCurry(forwardRefWithNameFn)

```
it's not perfect in that it doesn't fully type the forwardRef stuff, but at least you are getting type flow on T.  The safeCurry helper will flow generics if they are broken out seperately like this.

Reply:
Hmm... it doesn't seem to work due to the underlying `forwardRef` typing, and any permutation to coerce that inner typing makes the downstream components complain